Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -84,59 +91,70 @@ public Task<Calendar> GetCalendarByIdAsync(string externalId)
 		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
 		public Task<IList<CalendarEvent>> GetEventsAsync(Calendar calendar, DateTime start, DateTime end)
 		{
-			return Task.Run<IList<CalendarEvent>>(() =>
+			var calendars = new List<CalendarEvent>();
+			CalendarManager calManager = null;
+			CalendarQuery calQuery = null;
+			CalendarList calList = null;
+			CalendarRecord calRecord = null;
+			CalendarFilter calFilter = null;
+
+			try
 			{
-				var calendars = new List<CalendarEvent>();
-
-				CalendarManager calManager = new CalendarManager();
-				CalendarQuery calQuery = new CalendarQuery(Event.Uri);
-				CalendarList calList = calManager.Database.GetRecordsWithQuery(calQuery, 0, 0);
-				CalendarRecord calRecord = calList.GetCurrentRecord();
+				calManager = new CalendarManager();
+				calFilter.AddCondition(CalendarFilter.LogicalOperator.And, Event.Start, CalendarFilter.IntegerMatchType.GreaterThanOrEqual, ConvertIntPtrToCalendarTime(start));
+				calFilter.AddCondition(CalendarFilter.LogicalOperator.And, Event.End, CalendarFilter.IntegerMatchType.LessThanOrEqual, ConvertIntPtrToCalendarTime(end));
+				calQuery = new CalendarQuery(Event.Uri);
+				calQuery.SetFilter(calFilter);
+				calList = calManager.Database.GetRecordsWithQuery(calQuery, 0, 0);
 
 				if (calList.Count == 0)
-					return calendars;
+					return Task.FromResult<IList<CalendarEvent>>(calendars);
 
 				calList.MoveFirst();
+				calRecord = calList.GetCurrentRecord();
 
 				do
 				{
 					string summary = calRecord.Get<string>(Event.Summary);
-					int IsAllday = calRecord.Get<int>(Event.IsAllday);
-					CalendarTime sTime = Convert.ToBoolean(IsAllday) ? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)) : calRecord.Get<CalendarTime>(Event.Start);
-					CalendarTime End = Convert.ToBoolean(IsAllday) ? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)): calRecord.Get<CalendarTime>(Event.End);
+					CalendarTime startTime = calRecord.Get<CalendarTime>(Event.Start);
+					CalendarTime endTime = calRecord.Get<CalendarTime>(Event.End);
```

Comment:
You removed the `Convert.ToBoolean(IsAllday) ? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc))` bit from here, but kept it down below in the GetEventById version?? I'm just a bit confused by the inconsistency now..